graylog2-server/src/main/java/org/graylog2/plugin/inputs/transports/AbstractTcpTransport.java

```diff
@@ -269,8 +273,14 @@ public void stop() {
             try {
                 final SelfSignedCertificate ssc = new SelfSignedCertificate(configuration.getString(CK_BIND_ADDRESS) + ":" + configuration.getString(CK_PORT));
                 certFile = ssc.certificate();
-                keyFile = ssc.privateKey();
-            } catch (CertificateException e) {
+
+                if (!Strings.isNullOrEmpty(tlsKeyPassword)) {
+                    keyFile = KeyUtil.generatePKSC8PrivateKey(tlsKeyPassword.toCharArray(), ssc.key());
```

Comment:
It would be nice if we could replace the existing unencrypted key file.
Leaving the old unencrypted file in the temp directory provides no security benefit